tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestExampleGenerator.java

```diff
@@ -2076,4 +2079,51 @@ private String getResult(String starredPath, String attr) {
     public String sampleAttrAndValue(PathStarrer ps, final String separator, String value) {
         return ps.getAttributesString(separator) + "➔«" + value + "»";
     }
+
+    PathHeader.Factory phf = PathHeader.getFactory(null);
+
+    private boolean isValidPath(String xpath, ULocale locale) {
+        PathHeader ph = phf.fromPath(xpath);
+        if (ph == null) {
+            return false;
+        }
+        if (ph.getSurveyToolStatus() == PathHeader.SurveyToolStatus.DEPRECATED) {
+            return false;
+        }
+        if (ph.getSurveyToolStatus() == PathHeader.SurveyToolStatus.HIDE
+                || ph.getSurveyToolStatus() == PathHeader.SurveyToolStatus.READ_ONLY) {
+            return false;
+        }
+
+        if (SDI.getCoverageValue(xpath, locale.getBaseName())
+                > org.unicode.cldr.util.Level.COMPREHENSIVE.getLevel()) {
+            return false;
+        }
+        return true;
+    }
+
+    public void TestRefactoring() {
+        for (ULocale locale : ULocale.getAvailableLocales()) {
+            String loc = locale.toString();
```

Comment:
```suggestion
        for (final String loc : info.getCldrFactory().getAvailable()) {
```

Sorry, missed this before - slight change needed to make it run against all locales instead of just those ULocale thinks is available.

Reply:
but, not important since you're going to be removing this test